### PR TITLE
Support api v3.4

### PR DESCRIPTION
**Pull request by DavidMealha**
- Adds support to API v3.4
- Adds Ruby 3.1 to supported versions

### CODE REVIEW

File: spec/support/fixtures/applicant.json [hunk 1]
```diff
@@ -19,5 +19,9 @@
     "state":"",
     "postcode":"SW4 6EH",
     "country":"GBR"
+  },
```
Comment (Phoebe-B) [inline spec/support/fixtures/applicant.json:22]: You also need `phone_number` in here

File: spec/integrations/applicant_spec.rb [hunk 2]
```diff
@@ -23,6 +23,10 @@
         'town' => 'London',
         'postcode' => 'SW4 6EH',
         'country' => 'GBR'
+      },
```
Comment (Phoebe-B) [inline spec/integrations/applicant_spec.rb:26]: `phone_number` here as well

Comment (DavidMealha) [reply spec/integrations/applicant_spec.rb:26]: In this one is the request body, so it's optional

Comment (DavidMealha) [reply spec/integrations/applicant_spec.rb:26]: Updated!

File: spec/support/fixtures/applicant.json (hunk 1 above)
Comment (DavidMealha) [reply spec/support/fixtures/applicant.json:22]: Updated!

**Review by Phoebe-B — APPROVED** [review]

**Pull request merged**